addons/binding/org.openhab.binding.omnilink/src/main/java/org/openhab/binding/omnilink/handler/OmnilinkBridgeHandler.java

```diff
@@ -266,9 +257,8 @@ public Void call() throws Exception {
     }
 
     private void handleUnitStatus(UnitStatus stat) {
-        Integer number = stat.getNumber();
-        Thing theThing = unitThings.get(number);
-        logger.debug("received status update for unit: " + number + ", status: " + stat.getStatus());
+        Thing theThing = getChildThing(OmnilinkBindingConstants.THING_TYPE_UNIT, stat.getNumber());
+        logger.debug("received status update for unit: " + stat.getNumber() + ", status: " + stat.getStatus());
```

Comment:
just an FYI we should replace logging lines like this with something like:
`logger.debug("received status update for unit: {}, status: {} ", stat.getNumber() , stat.getStatus());`